enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,303 @@
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what ip pool the VM belongs 
+(using an annotation or searching for subnets) and reserve the address.
+
+#### Detecting hypershift VMs and live migration
+
+This enhancement only applies for hypershift kubevirt workers so the rest of 
+virtual machine should not use it, those VMs will be labeled with
+`kubevirt.io/allow-pod-network-bridge-binding-live-migration` by hypershift
+kubevirt provider and ovn-k will do point to point routing if detected, the of non
+hypershift VM should not add that special label.
+
+The VM's host node change after live migration and the traffic has to be 
+redirected over this host node. To do so the point to point routing has to be
+changed and to detect post live-migration event a pair of 
+kubevirt virt-launcher pod labels has to be inspected, `kubevirt.io/nodeName` 
+and `creationTimestamp`, the first one is set after live migration with the 
+name of the new host node and the second one allow ovn-k to disregard old
+virt-launcher pod (they are not deleted after live-migration).
+
+### Risks and Mitigations
+
+#### OVN extended arp_proxy features
+
+
+The current ovn arp_proxy is missing the following features: enforcing the mac 
+address and ipv6, ideally those get implemented and ovn-kubernetes have 
+them for 4.14 release.
+
+In case they don't arrive on time, it will not be able to configure IPv6 hosted
+clusters at hypershift and VMs arp table will need to be refreshed after live
+migration. 
+
+### Drawbacks
+
+Doing live migration over pod network with kubevirt and changing its topology may present some risks and need to be 
+thoroughly tested and supported.
+
+### Open Questions 
+
+Do this impact non hypershift payloads ? The control-plane will have some more
+load from the point to point routing but data-plane is not affected
+
+Do this work with future interconnect topology ? It's not using a distributed 
+switch per hosted cluster so that part should be fine, but it may needed 
+to replicate point to point routing at all node's ovn_cluster_router.
+
+### Test Plan
+
+Add some network tests to [hypershift conformance](https://github.com/openshift/origin/pull/27456)
+that will do live migration with the following scenarios:
+- TCP connection between VMs
+- TCP connection to VMs from service
+- TCP connection from VM to service
+
+The connection has to survive and downtime should be minimal.
+
+
+### Graduation Criteria
```

Comment:
Kubevirt provider is going GA for 4.14 but live migration will be dev preview for 4.14? Am I understanding this correctly.

Reply:
I have modify `Graduation Criteria` to clarify that

Reply:
I dont see it now.